sections/advanced/style-objects.md

```diff
@@ -24,3 +24,306 @@ render(
+      <DecendantBox/>
+    </Box>
+  </div>
+);
+```
+
+a styled-component can reference it's sibling(s) by a `className`.
+
+_child selector (>), adjacent sibling selector (+) and general sibling selector (~) work in the same way_
+
+```react
+
+// Adjacent Sibling selector by class
+const SiblingBox = styled.div({
+  backgroundColor: 'blue',
+  height: '50px',
+  width: '50px',
+  borderStyle: 'solid',
+  borderWidth: '4px',
+  borderColor: 'darkblue'
+  }
+)
+
+const Box = styled.div({
+    height: '50px',
+    width: '50px',
+    background: 'palevioletred',
+    '+ .sibling-box' : {
+      borderColor: 'dodgerblue'
+    }
+  }
+)
+
+render(
+  <div>
+    <Box />
+    <SiblingBox className='sibling-box'/>
+    <SiblingBox className='sibling-box'/>
+  </div>
+);
+```
+
+a styled-component can reference it's sibling(s) by component.
+
+_child selector (>), adjacent sibling selector (+) and general sibling selector (~) work in the same way_
+
+```react
+
+// Adjacent sibling selector by component
+const SiblingBox = styled.div({
+  backgroundColor: 'blue',
+  height: '50px',
+  width: '50px',
+  borderStyle: 'solid',
+  borderWidth: '4px',
+  borderColor: 'darkblue'
+  }
+)
+
+const Box = styled.div({
+    height: '50px',
+    width: '50px',
+    background: 'palevioletred',
+    [`+ ${SiblingBox}`] : {
+      borderColor: 'dodgerblue'
+    }
+  }
+)
+
+render(
+  <div>
+    <Box/>
+    <SiblingBox/>
+    <SiblingBox/>
+  </div>
+);
+```
+
+### Pseudo classes
+
+styled-components can reference pseudo classes and decendants with pseudo selectors
+
+```react
+const Label = styled.label({
+  position: 'relative',
+  cursor: 'pointer'
+})
+
+// Pseudo selector
+const Check = styled.span({
+  position: 'absolute',
+  top: '0px',
+  left: '0px',
+  height: '25px',
+  width: '25px',
+  borderWidth: '4px',
+  borderStyle: 'solid',
+  borderColor: 'dodgerblue',
+  background: 'white',
+  ':hover': {
+    borderColor: 'lightskyblue'
+  }
+})
+
+// Pseudo selector by component
+const Input = styled.input({
+    position: 'absolute',
+    opacity: 0,
+    height: '0px',
+    width: '0px',
+    [`:checked + ${Check}`] : {
+      background: 'blue'
+    }
+  }
+)
+
+render(
+  <div>
+  <Label>
+    <Input type='checkbox' />
+    <Check />
+  </Label>
+  </div>
+);
+```
+
+### Spread operator
+
+With style objects you can take advantage of JavaScript's spread syntax
+
+```react
+// JavaScript object for common styles
+const fontStyles = {
+  fontFamily: 'monospace',
+  fontSize: '1rem',
+  lineHeight: '1rem',
+  color: 'palevioletred',
+  margin: '0px',
+}
+
+// Spread common styles across components
+const H1 = styled.h1({
+  ...fontStyles,
+  fontSize: '2.5rem',
+  lineHeight: '2.5rem'
+})
+
+const P = styled.p({
+  ...fontStyles,
+  color: 'rgb(243, 182, 97)'
+})
+
+render(
+  <div>
+    <H1>h1. heading</H1>
+    <P>p. paragraph</P>
+  </div>
+);
+```
+
+### Unit-less css values
+
+styled-components object styles work with both numbers and string values
```

Comment:
`Style objects` work with both `number` and string values. Just like styled tagged template literals, number values are assumed as `px` unless used where unitless values are typically accepted such as `lineHeight` and `flexGrow`.